quisp/modules/QRSA/RuleEngine/RuleEngine.cc

```diff
@@ -990,13 +990,14 @@ void RuleEngine::updateResources_EntanglementSwapping(swapping_result swapr) {
     error("RuleEngine. Entanglement swapping went wrong");
   }
   // first delete old record
-  for (auto it = allResources[qnic_type][qnic_index].begin(); it != allResources[qnic_type][qnic_index].end(); ++it) {
+  auto &resources = allResources[qnic_type][qnic_index];
+  for (auto it = resources.begin(); it != resources.end(); ++it) {
     if (it->second == qubit) {
-      allResources[qnic_type][qnic_index].erase(it);
+      resources.erase(it++);
```

Comment:
this skips the next resource of the erased resource.
should be `resources.erase(it--);`

Reply:
Oh, yeah. Btw, in this case, actually, this must be called just once. So is it better to use break just after the race?

Reply:
yes, you should do `break`.